packages/wrangler/src/dev/dev-vars.ts

```diff
@@ -0,0 +1,38 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+import dotenv from "dotenv";
+import { logger } from "../logger";
+import type { Config } from "../config";
+
+/**
+ * Get the Worker `vars` bindings for a `wrangler dev` instance of a Worker.
+ *
+ * The `vars` bindings can be specified in the `wrangler.toml` configuration file.
+ * But "secret" `vars` are usually only provided at the server -
+ * either by creating them in the Dashboard UI, or using the `wrangler secret` command.
+ *
+ * It is useful during development, to provide these types of variable locally.
+ * When running `wrangler dev` we will look for a file called `.dev.vars`, situated
+ * next to the `wrangler.toml` file (or in the current working directory if there is no
+ * `wrangler.toml`).
+ *
+ * Any values in this file, formatted like a `dotenv` file, will add to or override `vars`
+ * bindings provided in the `wrangler.toml`.
+ */
+export function getVarsForDev(config: Config): Config["vars"] {
+  const configDir = path.resolve(path.dirname(config.configPath ?? "."));
```

Comment:
instead of insisting it be right next to wrangler.toml, how do you feel about just using `find-up`?

Reply:
This is related to the requirement that it be outside the current project?

Again, I think would we should keep this feature as limited as possible to begin with.

Perhaps an option would be to make this more configurable from the programmatic API that we have talked about?

Reply:
yeah let's not do the findup thing. but I do think we should let people set the path to the file.